enhancements/multi-arch/heterogenous-architecture.md

```diff
@@ -0,0 +1,270 @@
+
+### Implementation Details/Notes/Constraints
+
+Operators supporting multiple architectures cannot stop supporting a given architecture since it would break heterogenous clusters which are trying to upgrade
+
+### Risks and Mitigations
+
+Releases and upgrade graphs for all architectures have to move simultaneously and cannot be independent. If there is a bug on one arch, it will cause the edge or the release image to be removed for all arches
+
+## Design Details
+
+### RHCOS Support for ARM Azure
+
+The RHCOS pipeline will have to generate Azure disk images for ARM. However as of today, RHEL does not support hyper-v for ARM and that support is landing in RHEL8.6. Hence 4.11 builds of RHCOS for ARM Azure will be based on RHEL8.6.
+It is TBD whether a separate pipeline with early access RHCOS builds will be needed as RHEL8.6 would not have GA'ed before 4.11.
+
+### OCP Build & Release:
+
+Phase 0 will involve building and publishing of [nightly builds][OCPPLAN-7640]. The release payload will be a manifest listed image with the component images being manifest listed as well.
+All 4 architectures - x86, ARM, Power and Z will be included in the manifest list. Existing arch specific release payloads continue to be delivered through their individual release controllers.
+A new heterogeneous release controller is bootstrapped and displays "4-stable" entries for ART constructed release payloads.
+
+`oc adm release new` would be [enhanced][WRKLDS-370] to allow the creation of manifest list release payloads. The manifest list flow would be triggered whenever the CVO image in an imagestream was a manifest list.
+If the CVO image is a standard manifest, the generated release payload will also be a manifest. If the CVO image is a manifest list, the generated release payload would be a manifest list (containing a manifest
+for each arch possessed by the CVO manifest list). In either case, oc adm release new would permit non-CVO component images to be manifest or manifest lists and pass them through directly to the resultant release manifest(s).
+If a manifest list release payload is generated, each architecture specific release payload manifest will reference the same pullspecs provided in the input imagestream.
+
+Since the internal Openshift registry does not support manifest lists, the release controller would use quay.io. There will be a new architecture option added to Cincinnati called "multi"
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the single arch payload graph that we have today, [support][OCPPLAN-7642] needs to be added for an additional graph for hetero arch (called "multi" as noted above) which CVO will need to look at to
+make update recommendations. In order for CVO to know that the user wants to move to a heterogenous payload, a new field needs to be set in the cluster version resource or in the infrastructure resource for CVO to
+look at the hetero graph to make recommendations. There are a few upgrade scenarios to consider:
```

Comment:
as opposed to above, this might be an even better place to provide more details on why a new field is needed or what alternatives exist.

Reply:
updated this part.